src/metabase/server/handler.clj

```diff
@@ -1,26 +1,46 @@
-            [metabase.server.middleware.exceptions :as mw.exceptions]
-            [metabase.server.middleware.json :as mw.json]
-            [metabase.server.middleware.log :as mw.log]
-            [metabase.server.middleware.misc :as mw.misc]
-            [metabase.server.middleware.offset-paging :as mw.offset-paging]
-            [metabase.server.middleware.security :as mw.security]
-            [metabase.server.middleware.session :as mw.session]
-            [metabase.server.middleware.ssl :as mw.ssl]
-            [metabase.server.routes :as routes]
-            [ring.middleware.cookies :refer [wrap-cookies]]
-            [ring.middleware.gzip :refer [wrap-gzip]]
-            [ring.middleware.keyword-params :refer [wrap-keyword-params]]
-            [ring.middleware.params :refer [wrap-params]]))
+  (:require
+   [metabase.config :as config]
+   [metabase.server.middleware.auth :as mw.auth]
+   [metabase.server.middleware.browser-cookie :as mw.browser-cookie]
+   [metabase.server.middleware.exceptions :as mw.exceptions]
+   [metabase.server.middleware.json :as mw.json]
+   [metabase.server.middleware.log :as mw.log]
+   [metabase.server.middleware.misc :as mw.misc]
+   [metabase.server.middleware.offset-paging :as mw.offset-paging]
+   [metabase.server.middleware.security :as mw.security]
+   [metabase.server.middleware.session :as mw.session]
+   [metabase.server.middleware.ssl :as mw.ssl]
+   [metabase.server.routes :as routes]
+   [ring.core.protocols :as ring.protocols]
+   [ring.middleware.cookies :refer [wrap-cookies]]
+   [ring.middleware.gzip :refer [wrap-gzip]]
+   [ring.middleware.keyword-params :refer [wrap-keyword-params]]
+   [ring.middleware.params :refer [wrap-params]]))
 
-;; required here because this namespace is not actually used anywhere but we need it to be loaded because it adds
-;; impls for handling `core.async` channels as web server responses
-(classloader/require 'metabase.async.api-response)
+(extend-protocol ring.protocols/StreamableResponseBody
+  ;; java.lang.Double, java.lang.Long, and java.lang.Boolean will be given a Content-Type of "application/json; charset=utf-8"
+  ;; so they should be strings, and will be parsed into their respective values.
+  java.lang.Double
+  (write-body-to-stream [num response output-stream]
+    (ring.protocols/write-body-to-stream (str num) response output-stream))
+
+  java.lang.Long
+  (write-body-to-stream [num response output-stream]
+    (ring.protocols/write-body-to-stream (str num) response output-stream))
+
+  java.lang.Boolean
+  (write-body-to-stream [bool response output-stream]
+    (ring.protocols/write-body-to-stream (str bool) response output-stream))
+
+  clojure.lang.Keyword
+  (write-body-to-stream [kkey response output-stream]
+    (ring.protocols/write-body-to-stream
+     (if-let  [key-ns (namespace kkey)]
+       (str key-ns "/" (name kkey))
+       (name kkey))
+     response output-stream)))
```

Comment:
These were the only parts of `metabase.async.api-response` actually still needed. They have nothing to do with anything being async